### ISSUE TITLE
npm install Doesn't work

### ISSUE THREAD

**Issue by FeloVilches** [sjmelia/electron-boilerplate-sqlite#1]
Here's the log. The error message doesn't say anything anyway. The last line before the error is `Rebuilding native production dependencies for win32:x64`

```
Error: C:\Program Files\nodejs\node.exe exited with code 1
Output:

> sqlite3@3.1.8 install C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3
> node-pre-gyp install --fallback-to-build


C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )

C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64 )  else (node "" configure --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64 )

C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" build --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64 )  else (node "" build --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64 )
���̃\�����[�V�������̃v���W�F�N�g�� 1 �x�� 1 ���r���h���܂��B���s�r���h���L���ɂ����ɂ́A"/m" �X�C�b�`���ǉ����Ă��������B
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Platforms\x64\PlatformToolsets\v140\Toolset.targets(36,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\build\deps\action_before_build.vcxproj]
Failed to execute 'node-gyp.cmd build --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64' (1)

Error output:
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
gyp ERR! System Windows_NT 10.0.14393
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "build" "--fallback-to-build" "--module=C:\\Users\\Felo\\Desktop\\electron-boilerplate-sqlite-master\\node_modules\\sqlite3\\lib\\binding\\electron-v1.6-win32-x64\\node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=C:\\Users\\Felo\\Desktop\\electron-boilerplate-sqlite-master\\node_modules\\sqlite3\\lib\\binding\\electron-v1.6-win32-x64"
gyp ERR! cwd C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3
gyp ERR! node -v v6.2.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp.cmd build --fallback-to-build --module=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64\node_sqlite3.node --module_name=node_sqlite3 --module_path=C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\lib\binding\electron-v1.6-win32-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:106:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:850:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
node-pre-gyp ERR! System Windows_NT 10.0.14393
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Felo\\Desktop\\electron-boilerplate-sqlite-master\\node_modules\\sqlite3\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\sqlite3
node-pre-gyp ERR! node -v v6.2.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.31
node-pre-gyp ERR! not ok

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "rebuild" "sqlite3"
npm ERR! node v6.2.0
npm ERR! npm  v3.8.9
npm ERR! code ELIFECYCLE
npm ERR! sqlite3@3.1.8 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sqlite3@3.1.8 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the sqlite3 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs sqlite3
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls sqlite3
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\npm-debug.log

    at ChildProcess.childProcess.once.code (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\src\util\util.ts:136:14)
    at ChildProcess.g (events.js:286:16)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:850:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
From previous event:
    at spawn (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\src\util\util.ts:99:3)
    at C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\src\yarn.ts:146:11
From previous event:
    at rebuild (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\out\yarn.js:81:22)
    at C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\src\yarn.ts:15:11
    at next (native)
    at tryOnImmediate (timers.js:543:15)
    at processImmediate [as _immediateCallback] (timers.js:523:5)
From previous event:
    at installOrRebuild (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\out\yarn.js:33:21)
    at C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\src\cli\install-app-deps.ts:34:2
    at next (native)
From previous event:
    at main (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\out\cli\install-app-deps.js:36:21)
    at Object.<anonymous> (C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\node_modules\electron-builder\out\cli\install-app-deps.js:76:1)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Function.Module.runMain (module.js:575:10)
    at startup (node.js:160:18)
    at node.js:449:3

npm WARN optional Skipping failed optional dependency /7zip-bin/7zip-bin-mac:
npm WARN notsup Not compatible with your operating system or architecture: 7zip-bin-mac@1.0.1
npm WARN optional Skipping failed optional dependency /7zip-bin/7zip-bin-linux:
npm WARN notsup Not compatible with your operating system or architecture: 7zip-bin-linux@1.0.3
npm WARN electron-boilerplate-sqlite@1.0.0 No repository field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v6.2.0
npm ERR! npm  v3.8.9
npm ERR! code ELIFECYCLE
npm ERR! electron-boilerplate-sqlite@1.0.0 postinstall: `install-app-deps`
npm ERR! Exit status 4294967295
npm ERR!
npm ERR! Failed at the electron-boilerplate-sqlite@1.0.0 postinstall script 'install-app-deps'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the electron-boilerplate-sqlite package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     install-app-deps
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs electron-boilerplate-sqlite
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls electron-boilerplate-sqlite
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Felo\Desktop\electron-boilerplate-sqlite-master\npm-debug.log
```

**Comment by FeloVilches** [comment]
OK this actually works on Linux after installing some stuffs.

But I can't remember which packages I installed globally or locally to make it work.

**Comment by sjmelia** [comment]
Hi! Thanks for opening an issue.

In truth I haven't tried to build this on Windows - i'll give it a go. It looks like the problem is occurring (as we might expect) while we're trying to build SQLite; notice this error:

`C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Platforms\x64\PlatformToolsets\v140\Toolset.targets(36,5): error MSB8036: The Windows SDK version 8.1 was not found. Install the required version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution".`

There's probably something in the [Windows SDK 8.1](https://developer.microsoft.com/en-us/windows/downloads/windows-8-1-sdk) that you need to build it - you could try installing that and give it a go.

**Comment by silaspedrosa** [comment]
Yeah, same thing here. I'll try to install this windows 8.1 sdk and see what happens.

**Comment by sjmelia** [comment]
Hi guys sorry for the long absence.

I managed to successfully build on Win64. (Windows 10) In order to do this I had to install Visual Studio 2015 with the C++ libraries and a Python 2; pretty much following the instructions here: https://github.com/mapbox/node-sqlite3/wiki/Building-on-Windows

It was reasonably awkward (compared to the Linux build) but is definitely possible! I'm going to close the case.
